applications/PfemFluidDynamicsApplication/custom_utilities/move_mesh_utility.h

```diff
@@ -0,0 +1,71 @@
+//    |  /           |
+//    ' /   __| _` | __|  _ \   __|
+//    . \  |   (   | |   (   |\__ `
+//   _|\_\_|  \__,_|\__|\___/ ____/
+//                   Multi-Physics
+//
+//  License:		 BSD License
+//					 Kratos default license: kratos/license.txt
+//
+//  Main authors:    Klauss B Sautter
+//                   Alejandro Cornejo Velazquez
+//                   Carlos Eulogio Flores
+//
+//
+
+#ifndef KRATOS_PFEM_MOVE_MESH_UTILITY_H_INCLUDED
+#define KRATOS_PFEM_MOVE_MESH_UTILITY_H_INCLUDED
+
+
+// System includes
+
+
+// External includes
+
+
+// Project includes
+#include "includes/define.h"
+#include "includes/model_part.h"
+
+
+namespace  Kratos
+{
+
+    /**
+     * @class MoveMeshUtility
+     *
+     * @brief This utilitiy moves the nodes of a pfem fluid
+     *        used especially in the pfem-fem coupling
+     *
+     * @author Klaus B Sautter
+     */
+    class  KRATOS_API(PFEM_FLUID_DYNAMICS_APPLICATION) MoveMeshUtility
+    {
+        public:
+            typedef ModelPart::NodesContainerType NodesContainerType;
+
+            KRATOS_CLASS_POINTER_DEFINITION(MoveMeshUtility);
+
+            MoveMeshUtility() {};
+            ~MoveMeshUtility() {};
```

Comment:
```suggestion
            MoveMeshUtility() = delete;
            ~MoveMeshUtility() = default;
```

Reply:
I tried this (with the `static` modification you suggest below in both the header and python binding), but I get this compilation error, which makes me think that pybind11 tries to use the deleted constructor. I assume you suggested this to prevent the usage of this class without passing any arguments to it. To circumvent this, I did `MoveMeshUtility() = default`, which compiles but this replicates the older behavior. Do you have any other hint about this error?


```
In file included from /home/sose2021/Software/Kratos/external_libraries/pybind11/pybind11.h:48,
                 from /home/sose2021/Software/Kratos/applications/PfemFluidDynamicsApplication/custom_python/add_custom_utilities_to_python.h:14,
                 from /home/sose2021/Software/Kratos/applications/PfemFluidDynamicsApplication/custom_python/add_custom_utilities_to_python.cpp:15:
/home/sose2021/Software/Kratos/external_libraries/pybind11/detail/init.h: In instantiation of ‘Class* pybind11::detail::initimpl::construct_or_initialize(Args&& ...) [with Class = Kratos::MoveMeshUtility; Args = {}; typename std::enable_if<(! std::is_constructible<_Tp, _Args>::value), int>::type <anonymous> = 0]’:
/home/sose2021/Software/Kratos/external_libraries/pybind11/detail/init.h:175:66:   required from ‘static void pybind11::detail::initimpl::constructor<Args>::execute(Class&, const Extra& ...) [with Class = pybind11::class_<Kratos::MoveMeshUtility, std::shared_ptr<Kratos::MoveMeshUtility> >; Extra = {}; typename std::enable_if<(! Class::has_alias), int>::type <anonymous> = 0; Args = {}]’
/home/sose2021/Software/Kratos/external_libraries/pybind11/pybind11.h:1370:9:   required from ‘pybind11::class_<type_, options>& pybind11::class_<type_, options>::def(const pybind11::detail::initimpl::constructor<Args ...>&, const Extra& ...) [with Args = {}; Extra = {}; type_ = Kratos::MoveMeshUtility; options = {std::shared_ptr<Kratos::MoveMeshUtility>}]’
/home/sose2021/Software/Kratos/applications/PfemFluidDynamicsApplication/custom_python/add_custom_utilities_to_python.cpp:48:26:   required from here
/home/sose2021/Software/Kratos/external_libraries/pybind11/detail/init.h:63:64: error: use of deleted function ‘Kratos::MoveMeshUtility::MoveMeshUtility()’
   63 | inline Class *construct_or_initialize(Args &&...args) { return new Class{std::forward<Args>(args)...}; }
      |                                                                ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
In file included from /home/sose2021/Software/Kratos/applications/PfemFluidDynamicsApplication/custom_python/add_custom_utilities_to_python.cpp:25:
/home/sose2021/Software/Kratos/applications/PfemFluidDynamicsApplication/custom_utilities/move_mesh_utility.h:49:13: note: declared here
   49 |             MoveMeshUtility() = delete;
      |             ^~~~~~~~~~~~~~~
make[2]: *** [applications/PfemFluidDynamicsApplication/CMakeFiles/KratosPfemFluidDynamicsApplication.dir/build.make:102: applications/PfemFluidDynamicsApplication/CMakeFiles/KratosPfemFluidDynamicsApplication.dir/custom_python/add_custom_utilities_to_python.cpp.o] Error 1
make[1]: *** [CMakeFiles/Makefile2:2516: applications/PfemFluidDynamicsApplication/CMakeFiles/KratosPfemFluidDynamicsApplication.dir/all] Error 2
make: *** [Makefile:130: all] Error 2

```

Reply:
It is also worth noting that I removed the `MoveMeshUtility::MovePfemMesh` function as it is already implemented in the `VariableUtils::UpdateCurrentPosition`. After that removal the above compilation error happened.

Reply:
Can you also my suggestion in the python interface?
Then this error should be gone